python/ray/tune/tests/test_tune_restore.py

```diff
@@ -186,6 +188,20 @@ def on_step_begin(self, iteration, trials, **kwargs):
                 assert len(trials) == self.expected_trials
                 self._checked = True
 
+    class CheckTrialResourcesCallback(Callback):
+        """Checks if unfinished trials are resumed with updated resources."""
+
+        def __init__(self, expected_new_cpu: int):
+            self._expected_new_cpu = expected_new_cpu
+            self._checked = False
+
+        def on_step_begin(self, iteration: int, trials: List["Trial"], **info):
+            if not self._checked:
+                for trial in trials:
+                    if trial.status == Trial.PENDING:
+                        assert trial.resources.cpu == self._expected_new_cpu
+                self._checked = True
```

Comment:
```suggestion
        def __init__(self, expected_new_cpu: int, check_after: int = 20):
            self._expected_new_cpu = expected_new_cpu
            self._check_after = check_after
            self._checked = False

        def on_step_begin(self, iteration: int, trials: List["Trial"], **info):
            if not self._checked and iteration >= self._check_after:
                for trial in trials:
                    if trial.status == Trial.PENDING:
                        assert trial.resources.cpu == self._expected_new_cpu
                    self._checked = True
```

This ensures that the code path is reached at least once, and only after a certain amount of step iterations. This shouldn't make a difference for the restored trial runner (as the Trial objects are recreated) but will make a difference if we add the callback to the first check (with the failure callback), as trials will not be generated, yet, on the first call of this callback.